### PR TITLE
samples: Bluetooth: Add ISO Broadcaster and Synchronized Receiver sample 

### PR DESCRIPTION
**Pull request by cvinayak**
- Add an application that initializes Bluetooth Subsystem,
creates an Extended Advertising set, starts Periodic
Advertising and creates BIG to demonstrate ISO Broadcaster
functionality.

- Add an application that initializes Bluetooth Subsystem,
starts scanning for Periodic Advertising, establishes
Periodic Advertising Sync, receives Periodic Advertising
reports, receives BIGInfo reports and creates BIG sync.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: samples/bluetooth/iso_broadcast/prj.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+CONFIG_BT_EXT_ADV=y
+CONFIG_BT_PER_ADV=y
+CONFIG_BT_ISO=y
+CONFIG_BT_ISO_BROADCAST=y
+CONFIG_BT_DEBUG_LOG=y
+CONFIG_BT_DEVICE_NAME="Test ISO Broadcaster"
+
+# Temporary, enable the following to meet BT_ISO dependencies
+CONFIG_BT_OBSERVER=y
+CONFIG_BT_PER_ADV_SYNC=y
```
Comment (Thalley) [inline samples/bluetooth/iso_broadcast/prj.conf:11]: `CONFIG_BT_ISO_BROADCAST` `select`s `CONFIG_BT_PER_ADV_SYNC` so this shouldn't be necessary.

File: samples/bluetooth/iso_broadcast/prj.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
+CONFIG_BT=y
+CONFIG_BT_EXT_ADV=y
+CONFIG_BT_PER_ADV=y
+CONFIG_BT_ISO=y
+CONFIG_BT_ISO_BROADCAST=y
+CONFIG_BT_DEBUG_LOG=y
+CONFIG_BT_DEVICE_NAME="Test ISO Broadcaster"
+
+# Temporary, enable the following to meet BT_ISO dependencies
+CONFIG_BT_OBSERVER=y
```
Comment (Thalley) [inline samples/bluetooth/iso_broadcast/prj.conf:10]: Why is this needed?

File: samples/bluetooth/iso_broadcast/sample.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+sample:
+  name: Bluetooth ISO Broadcaster
+tests:
+  sample.bluetooth.iso_broadcast:
+    harness: bluetooth
+    platform_allow: qemu_cortex_m3 qemu_x86 nrf52_bsim nrf52dk_nrf52832
```
Comment (Thalley) [inline samples/bluetooth/iso_broadcast/sample.yaml:6]: How about the nrf53 actually?

File: samples/bluetooth/iso_broadcast/src/main.c [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+		mfg_data[2]++;
+
+		printk("Set Periodic Advertising Data...");
+		err = bt_le_per_adv_set_data(adv, ad, ARRAY_SIZE(ad));
+		if (err) {
+			printk("failed (err %d)\n", err);
+			return;
+		}
+		printk("done.\n");
```
Comment (Thalley) [inline samples/bluetooth/iso_broadcast/src/main.c:139]: Why does the broadcaster set periodic advertising data? 

I suppose it should broadcast ISO data instead using `bt_iso_chan_send` instead

File: samples/bluetooth/iso_receive/README.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+* A board with Bluetooth Low Energy support
+
+Building and Running
+********************
+
+This sample can be found under :zephyr_file:`samples/bluetooth/iso_receive` in
+the Zephyr tree.
+
+Use the sample found under :zephyr_file:`samples/bluetooth/iso_broadcasr` on
```
Comment (Thalley) [inline samples/bluetooth/iso_receive/README.rst:23]:
```suggestion
Use the sample found under :zephyr_file:`samples/bluetooth/iso_broadcast` on
```

**Review by Thalley — CHANGES_REQUESTED** [review]
Generally looks good. One major comment regarding the actual iso data.

File: samples/bluetooth/iso_broadcast/prj.conf (hunk 1 above)
Comment (cvinayak) [reply samples/bluetooth/iso_broadcast/prj.conf]: Yes, this can be removed.

File: samples/bluetooth/iso_broadcast/sample.yaml (hunk 3 above)
Comment (cvinayak) [reply samples/bluetooth/iso_broadcast/sample.yaml:6]: It can be added when ISO controller support is functional upstream, as there is need for board specific conf required to match corresponding controller used over rpmsg transport.

File: samples/bluetooth/iso_broadcast/src/main.c (hunk 4 above)
Comment (cvinayak) [reply samples/bluetooth/iso_broadcast/src/main.c:139]: This is to test periodic advertising data changes while maintaining the BIGInfo. `bt_iso_chan_send` is not added to this sample yet. 

Feel free to include it here and send a commit to this PR (Zephyr controller does not have data path hooked yet to test myself the data).

File: samples/bluetooth/iso_broadcast/prj.conf (hunk 2 above)
Comment (Thalley) [reply samples/bluetooth/iso_broadcast/prj.conf:10]: Right. The issue is that iso broadcast support for receiver requires this and we only have a aggregated Kconfig for iso broadcast support. There's a TODO to split it into broadcaster and receiver, so I guess that needs to be done before we can fix these :) I guess that's on me ^_^

File: samples/bluetooth/iso_broadcast/prj.conf (hunk 1 above)
Comment (Thalley) [reply samples/bluetooth/iso_broadcast/prj.conf]: See https://github.com/zephyrproject-rtos/zephyr/pull/34219#discussion_r611581592

File: samples/bluetooth/iso_broadcast/src/main.c (hunk 4 above)
Comment (Thalley) [reply samples/bluetooth/iso_broadcast/src/main.c:139]: Hmm, I guess it makes sense to test that, but does that belong in a sample application? It shouldn't be necessary for the application to set any PA data, and thus I suggest we remove that, or at least, add a todo to remove it and once we have data path then remove it. 

Imo setting PA data doesn't belong in a ISO broadcast sample.

**Comment by Thalley** [comment]
@cvinayak and reviewers: Just FYI: I will be making some changes to the commits based on some additional testing, as well as adding code to actually send ISO data.

File: samples/bluetooth/iso_receive/src/main.c [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+			return;
+		}
+		printk("success.\n");
+
+		printk("Creating Periodic Advertising Sync...");
+		bt_addr_le_copy(&sync_create_param.addr, &per_addr);
+		sync_create_param.options = 0;
+		sync_create_param.sid = per_sid;
+		sync_create_param.skip = 0;
+		sync_create_param.timeout = per_interval_ms * 2;
```
Comment (cvinayak) [inline samples/bluetooth/iso_receive/src/main.c:348]: The timeout is in 10 ms units, right?

```suggestion
		sync_create_param.timeout = per_interval_ms * 6 / 10 ;
```

Comment (Thalley) [reply samples/bluetooth/iso_receive/src/main.c]: Correct. Also added a #define for the multiplier if we want to use the same for both this timeout and the sem timeout.

**Review by Thalley — APPROVED** [review]
LGTM

**Review by asbjornsabo — APPROVED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]
